File: tutorials/testing/c++/eq_snippet.cpp

#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/EarthquakeUSGS.h"


using namespace bridges;

int max_quakes = 100;

// This program fragment illustrates how to access the USGS earthquake data
int main(int argc, char **argv) {

	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");
	// set title
	bridges.setTitle("Accessing USGIS Earthquake Data (USGIS Data)");

	// read the earth quake  data
	DataSource ds (&bridges);
	vector<EarthquakeUSGS> eq_list = ds.getEarthquakeUSGSData(max_quakes);

	// print the first quake record
	for (int k = 0; k < 10; k++) {
		cout << "Earthquake " << k << ": \n";
		cout << "\tMagnitude:" << eq_list[0].getMagnitude() << endl
		<< "\tDate:" << eq_list[0].getDateStr() << endl
		<< "\tLocation: " <<  eq_list[0].getLocation() << endl
		<< "\tLat/Long:"  << eq_list[0].getLatit() << "," <<
		eq_list[0].getLongit() << endl;
	}


	return 0;
}

File: tutorials/testing/java/eq_snippet.java
import java.util.List;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.EarthquakeUSGS;

// This program fragment illustrates how to access the USGS earthquake data
public class eq_snippet {
	public static void main(String[] args) throws Exception {

		// create Bridges object
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
		// read the earth quake  data and build the BST
		bridges.setTitle("Accessing USGIS Earthquake Data (USGS Data)");

		DataSource ds = bridges.getDataSource();

		// get 100 most recent earthquake records
		List<EarthquakeUSGS> eq_list = ds.getEarthquakeUSGSData(100);

		// print the first record
		System.out.println("Earthquake 0:");
		System.out.println ("\tMagnitude:"  + eq_list.get(0).getMagnitude() + "\n"
			+ "\tDate:" + eq_list.get(0).getTime() + "\n"
			+ "\tLocation: " +  eq_list.get(0).getLocation() + "\n"
			+ "\tLat/Long:"  + eq_list.get(0).getLatit() + "," +
			eq_list.get(0).getLongit());
	}
}
